src/main/scala/extractors/Extractor.scala

```diff
@@ -0,0 +1,50 @@
+package extractors
+
+object Extractor {
+
+  val inUaStr: String = "in.ua"
+  val gmailComStr: String = "gmail.com"
+
+  def filterNotInUa(domains: List[String]): List[String] = {
+    domains.filterNot {
+      case Domain(f, s, _*) if s"$s.$f" == inUaStr => true
+      case _ => false
+    }
+  }
+
+  def filterGmail(emails: Array[String]): Array[Unit] = {
```

Comment:
Function name `filterGmail` does not represent the actual function behavior.

Reply:
Function name has been changed to filterEmailsWithParticularDomain. Added optional parameter inputDomainStr: String with default value
filterNotInUa function name has been changed as well. Now it is filterNotDomainsWithParticularHead with optional String parameter

Reply:
What I meant was that your function prints email addresses that are not filtered. A more appropriate name would be something like `printEmailsFilteredByDomain`

Reply:
Function name has been changed